macro/src/codegen.rs

```diff
@@ -124,7 +125,20 @@ impl SpannedClassInfo {
                 unsafe impl<#(#java_class_generics,)*> JavaObject for #struct_name<#(#java_class_generics,)*>
                 where
                     #(#java_class_generics: JavaObject,)*
-                {}
+                {
+                    fn class<'jvm>(jvm: &mut Jvm<'jvm>) -> duchess::Result<'jvm, &'static Global<java::lang::Class>> {
+                        #cached_class
+                    }
+                }
+
+                unsafe impl<#(#java_class_generics,)*> ArrayElement for #struct_name<#(#java_class_generics,)*>
```

Comment:
I feel like I changed how this works...merge conflict?

Reply:
I think this was on an older commit I didn't squash after I rebased. Were you looking at the final diff?